pinecone/index.py

```diff
@@ -108,34 +291,88 @@ def _query_transform(item):
             ),
             **{k: v for k, v in kwargs.items() if k in _OPENAPI_ENDPOINT_PARAMS}
         )
-        return parse_query_response(response, vector or id)
+        return parse_query_response(response, vector is not None or id)
 
     @validate_and_convert_errors
-    def update(self, id, **kwargs):
+    def update(self,
+               id: str,
+               values: Optional[List[float]] = None,
+               set_metadata: Optional[Dict[str, Any]] = None,
+               namespace: Optional[str] = None,
+               **kwargs) -> Dict[str, Any]:
+        """
+        The Update operation updates vector in a namespace.
+        If a value is included, it will overwrite the previous value.
+        If a set_metadata is included,
+        the values of the fields specified in it will be added or overwrite the previous value.
+
+        API reference: https://docs.pinecone.io/reference/update
+
+        Args:
+            id (str): Vector's unique id.
+            values (List[float]): vector values to set. [optional]
+            set_metadata (Dict[str, Union[bool, date, dict, float, int, list, str, none_type)]]):
+                metadata to set for vector. [optional]
+            namespace (str): Namespace name where to update the vector.. [optional]
+
+        Keyword Args:
+        Supports OpenAPI client keyword arguments. See pinecone.core.client.models.UpdateRequest for more details.
+
+        Returns: An empty dictionary if the update was successful.
+
+        Call Examples:
+            index.update(id='id1', values=[1, 2, 3], namespace='my_namespace')
+            index.update(id='id1', set_metadata={'key': 'value'}, namespace='my_namespace')
+        """
         _check_type = kwargs.pop('_check_type', False)
+        args_dict = self._parse_args_to_dict([('values', values),
+                                              ('set_metadata', set_metadata),
+                                              ('namespace', namespace)])
         return self._vector_api.update(UpdateRequest(
                 id=id,
+                **args_dict,
                 _check_type=_check_type,
                 **{k: v for k, v in kwargs.items() if k not in _OPENAPI_ENDPOINT_PARAMS}
             ),
             **{k: v for k, v in kwargs.items() if k in _OPENAPI_ENDPOINT_PARAMS})
 
     @validate_and_convert_errors
-    def describe_index_stats(self, *args, **kwargs):
+    def describe_index_stats(self,
+                             filter: Dict[str, Any] = {},
+                             **kwargs) -> DescribeIndexStatsResponse:
+        """
+        The DescribeIndexStats operation returns statistics about the index's contents.
+        For example: The vector count per namespace and the number of dimensions.
+
+        API reference: https://docs.pinecone.io/reference/describe_index_stats_post
+
+        Args:
+            filter (Dict[str, Union[bool, date, dict, float, int, list, str, none_type)]]):
```

Comment:
For the docstring purposes, don't you think it will be more useful to put it like this even if it's not a proper type? BTW so what is the type of the date?

Reply:
Supported types https://docs.pinecone.io/docs/metadata-filtering#supported-metadata-types